### DISCUSSION TITLE
OCI_Execute in C loop

### DISCUSSION THREAD

**Discussion by aravindsreeram062** [vrogier/ocilib#312] [Q&A]
I'm trying to insert data into a table by calling Pl/Sql procedure inside a C loop.
Below Pl/Sqlcall I'm making 
`char* emp_ins = strdup("BEGIN ins_pack.insert_to_emp(:emp_id_out); END;");`

```
if (OCI_Prepare(st,emp_ins ) == FALSE)
    {
       trace("Query Statement: <%s>",emp_ins );
       trace("Error in OCI_Prepare");
       OCI_StatementFree(st);
       free(emp_ins );
       return 1;
    }
```
I want to run the insert statement in loop 10 times, is this correct way to execute in a loop? or any other better way we have? please suggest.
```
while(i<=10)
{
    if(OCI_Execute(st) == FALSE)
    {
         trace("OCI execution failed...");
           OCI_StatementFree(st);
          free(emp_ins );
          return 1;
     }
printf("emp_id_out <%d>",emp_id_out);
i++;
}
```

**Comment by vrogier** [comment]
Hi,

You an use a loop indeed.
You can also use bulk operations, e.g. binding an array of value that can be either processed in a loop within the PL/SQL block or modify the stored proc to to take an array.
Passing host arrays to either the PL/SQL block or procedure will reduce the number of network round-trips.
In your example, we could have only one network round-trip between your app the the database rather having 10.

Regards,

Vincent
